lib/Ray/sources/Window.hpp

```diff
@@ -111,9 +112,13 @@ namespace RAY {
 			void setTitle(const std::string &title);
 
 
-			//! @brief draw rectangle
+			//! @brief draw drawable
 			//! @param drawable The drawable to render on screen
-			void draw(Drawables::IDrawable &drawable) override;
```

Comment:
Why did you change this? You should take the lowest type that fit your needs as parameters, here a IDrawable is enough since they all have a drawOn(Window) method. There is no need to specify a specific method for 2D and 3D.